mloop/neuralnet.py

```diff
@@ -92,34 +87,40 @@ def __init__(self,
 
         with self.graph.as_default():
             ## Inputs
-            self.input_placeholder = tf.placeholder(tf.float32, shape=[None, self.num_params])
-            self.output_placeholder = tf.placeholder(tf.float32, shape=[None, 1])
-            self.keep_prob_placeholder = tf.placeholder_with_default(1., shape=[])
-            self.regularisation_coefficient_placeholder = tf.placeholder_with_default(0., shape=[])
+            self.input_placeholder = tf.compat.v1.placeholder(
+                tf.float32, shape=[None, self.num_params])
```

Comment:
this is just a question, would this inconsistency in dtype cause any trouble if calling NN from qctrl-mloop since we are using 64 there?

Reply:
I don't think there'll be any issues, because the M-LOOP neural net learner never needs to talk to the `qctrl-mloop` learners (even if they did get used together, there are separate graph objects involved, so I don't think we'd ever be mixing tensors from one to the other).

Reply:
Thanks, that makes sense.

Reply:
Oh, and as for any inconsistency when converting between NumPy data and tensors, I guess it could lead to some loss of precision (e.g. if the user is using float64 for costs and parameters), but I don't think it will be a big deal. I think the double precision is mainly useful for the internals of computations involving messy operations like division and whatnot; for the beautiful and smooth things we use in the neural network I can't imagine any issues arising.